### PR TITLE
Auth: Validation fixes for SSO Settings

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
Add URL validations for Auth URL, Token URL + Teams URL (for Generic OAuth).

**Why do we need this feature?**

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/login/social/connectors/grafana_com_oauth.go [hunk 1]
```diff
@@ -64,7 +64,9 @@ func (s *SocialGrafanaCom) Validate(ctx context.Context, settings ssoModels.SSOS
 		return err
 	}
 
-	// add specific validation rules for GrafanaCom
+	if info.AuthUrl != "" || info.TokenUrl != "" || info.TeamsUrl != "" {
```
Comment (dmihai) [inline pkg/login/social/connectors/grafana_com_oauth.go:67]: Should we use the validator funcs we defined for the other providers? Similar to `validation.MustBeEmptyValidator(info.AuthUrl, "Auth URL")`.

File: public/app/features/auth-config/utils/url.ts [hunk 2]
```diff
@@ -4,3 +4,12 @@ import { AuthProviderInfo } from '../types';
 export function getProviderUrl(provider: AuthProviderInfo) {
   return BASE_PATH + (provider.configPath || provider.id);
 }
+
+export const isUrlValid = (url: string): boolean => {
+  try {
+    const parsedUrl = new URL(url);
+    return parsedUrl.protocol.includes('http');
```
Comment (dmihai) [inline public/app/features/auth-config/utils/url.ts:11]: I'm not sure whether we should be very strict here. I cannot think of any scenarios where this condition might block legit urls, but who knows. I would just leave the `new URL(url)` for validation on client side, but up to you.

**Review by dmihai — APPROVED** [review]

**Review by dmihai — APPROVED** [review]

File: pkg/login/social/connectors/grafana_com_oauth.go (hunk 1 above)
Comment (mgyongyosi) [reply pkg/login/social/connectors/grafana_com_oauth.go]: Thanks, done!

File: public/app/features/auth-config/fields.tsx [hunk 3]
```diff
@@ -214,6 +229,18 @@ export function fieldMap(provider: string): Record<string, FieldData> {
       ),
       validation: {
         required: false,
+        validate: (value) => {
+          if (typeof value !== 'string') {
+            return false;
+          }
```
[the remark this answers is not in the document]
Comment (mgyongyosi) [reply public/app/features/auth-config/fields.tsx:229]: Thanks, done!

File: public/app/features/auth-config/utils/url.ts (hunk 2 above)
Comment (mgyongyosi) [reply public/app/features/auth-config/utils/url.ts:14]: I have the same on the backend as well, so I'd prefer to leave it like this and make sure that scheme is either http or https.

**Pull request merged**